delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/serialization/CompositeViewFieldsJsonLdSourceDecoder.scala

```diff
@@ -0,0 +1,51 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.serialization
+
+import ch.epfl.bluebrain.nexus.delta.kernel.utils.UUIDF
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.model.{contexts, CompositeViewFields, CompositeViewRejection}
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.RemoteContextResolution
+import ch.epfl.bluebrain.nexus.delta.rdf.syntax.jsonOpsSyntax
+import ch.epfl.bluebrain.nexus.delta.sdk.jsonld.JsonLdSourceProcessor.JsonLdSourceDecoder
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.Project
+import io.circe.Json
+import io.circe.syntax._
+import monix.bio.IO
+
+/**
+  * Decoder for [[CompositeViewFields]] which maps some fields to string, before decoding to get around lack of support
+  * for @json in json ld library.
+  */
+final class CompositeViewFieldsJsonLdSourceDecoder private (
```

Comment:
This approach seems a little cleaner than what we have in ElasticSearchViews.
Maybe is not necessary in the current PR, but can you right after make the same change in `ElasticSearchViews` (having its own JsonLdSourceDecoder class and having a similar way to map keys to json) ?